tests/tt_metal/tt_metal/unit_tests_fast_dispatch_single_chip_multi_queue/command_queue/test_EnqueueTrace.cpp

```diff
@@ -94,20 +89,160 @@ TEST_F(MultiCommandQueueSingleDeviceFixture, EnqueueOneProgramTrace) {
         input_data[i] = i;
     }
 
-    EnqueueWriteBuffer(data_movement_queue, input, input_data.data(), true);
-    Trace trace = BeginTrace(trace_queue);
-    EnqueueProgram(trace_queue, simple_program, false, trace);
-    Finish(trace_queue);
-    EndTrace(trace);
+    // Eager mode
     vector<uint32_t> eager_output_data;
     eager_output_data.resize(input_data.size());
+
+    EnqueueWriteBuffer(data_movement_queue, input, input_data.data(), true);
+    EnqueueProgram(command_queue, simple_program, true);
     EnqueueReadBuffer(data_movement_queue, output, eager_output_data.data(), true);
 
+    // Trace mode
     vector<uint32_t> trace_output_data;
     trace_output_data.resize(input_data.size());
-    EnqueueTrace(trace, true);
+
+    Trace trace;
+    EnqueueWriteBuffer(data_movement_queue, input, input_data.data(), true);
+
+    BeginTrace(trace);
+    EnqueueProgram(trace.trace_queue(), simple_program, false);
```

Comment:
Nitpick: given the trace object type is trace, I don't think it's member function needs to be called 'trace_queue'. Maybe just 'queue'

Reply:
i thought about this too, it looks redundant because the object is called trace too. But I don't have any strong preferences so will update